core/src/main/java/org/kiwix/kiwixmobile/core/main/ToolbarScrollingKiwixWebView.java

```diff
@@ -32,15 +32,27 @@
   private final int toolbarHeight = DimenUtils.getToolbarHeight(getContext());
   private final View toolbarView;
   private final View bottomBarView;
+  private final View parentNavigationBar;
   private final SharedPreferenceUtil sharedPreferenceUtil;
   private float startY;
 
+  public ToolbarScrollingKiwixWebView(Context context, WebViewCallback callback, AttributeSet attrs,
+    ViewGroup nonVideoView, ViewGroup videoView, CoreWebViewClient webViewClient, View toolbarView,
+    View bottomBarView, View parentNavigationBar, SharedPreferenceUtil sharedPreferenceUtil) {
+    super(context, callback, attrs, nonVideoView, videoView, webViewClient);
+    this.toolbarView = toolbarView;
+    this.bottomBarView = bottomBarView;
+    this.parentNavigationBar = parentNavigationBar;
+    this.sharedPreferenceUtil = sharedPreferenceUtil;
+  }
+
   public ToolbarScrollingKiwixWebView(Context context, WebViewCallback callback, AttributeSet attrs,
     ViewGroup nonVideoView, ViewGroup videoView, CoreWebViewClient webViewClient, View toolbarView,
     View bottomBarView, SharedPreferenceUtil sharedPreferenceUtil) {
     super(context, callback, attrs, nonVideoView, videoView, webViewClient);
```

Comment:
you should invoke the other constructor with `this` so you don't repeat simple mappings that could introduce subtle bugs. Not very relevant in kotlin because it is an actually decent language